shuffler/src/main/java/com/shuffle/p2p/OtrChannel.java

```
@@ -0,0 +1,585 @@
+
+            public OtrSession(Session<Address, Bytestring> session) {
+                this.session = session;
+            }
+
+            @Override
+            public synchronized boolean send(Bytestring message) {
+                try {
+                    sendClient.send("recipient",message);
+                    return true;
+                } catch (OtrException e) {
+                    return false;
+                } catch (InterruptedException er) {
+                    return false;
+                }
+            }
+
+            @Override
+            public synchronized void close() {
+                try {
+                    session.close();
+                } catch (InterruptedException e) {
+                    return;
+                }
+            }
+
+            @Override
+            public synchronized boolean closed() {
+                try {
+                    return session.closed();
+                } catch (InterruptedException e) {
+                    return false;
+                }
+            }
+
+            // TODO
+            @Override
+            public Peer<Address, Bytestring> peer() {
+                return OtrPeer.this;
+            }
+
+        }
+
+    }
+
+    private class OtrConnection implements Connection<Address> {
+
+        Connection<Address> connection;
+
+        public OtrConnection(Connection<Address> connection) {
+            this.connection = connection;
+        }
+
+        @Override
+        public void close() {
+            synchronized (lock) {
+                running = false;
+                try {
+                    this.connection.close();
+                } catch (InterruptedException e) {
+                    running = true;
+                }
+            }
+        }
+
+        @Override
+        public boolean closed() {
+            try {
+                return connection.closed();
+            } catch (InterruptedException e) {
```

Don't catch! Just let it throw.